internal/util/errors.go

```diff
@@ -16,98 +16,46 @@ limitations under the License.
-func (e ErrObjectExists) Unwrap() error {
-	return e.err
-}
-
-// ErrObjectNotFound is returned when named omap is not found in rados
-type ErrObjectNotFound struct {
-	oMapName string
-	err      error
-}
-
-// Error returns the error string for ErrObjectNotFound.
-func (e ErrObjectNotFound) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrObjectNotFound) Unwrap() error {
-	return e.err
-}
-
-// ErrSnapNameConflict is generated when a requested CSI snap name already exists on RBD but with
-// different properties, and hence is in conflict with the passed in CSI volume name
-type ErrSnapNameConflict struct {
-	requestName string
-	err         error
-}
+import (
+	"errors"
+	"fmt"
+)
 
-// Error returns the error string for ErrSnapNameConflict.
-func (e ErrSnapNameConflict) Error() string {
-	return e.err.Error()
-}
-
-// Unwrap returns the encapsulated error.
-func (e ErrSnapNameConflict) Unwrap() error {
-	return e.err
-}
+var (
+	// ErrKeyNotFound is returned when requested key in omap is not found
+	ErrKeyNotFound = errors.New("key not found")
+	// ErrObjectExists is returned when named omap is already present in rados
+	ErrObjectExists = errors.New("object exists")
+	// ErrObjectNotFound is returned when named omap is not found in rados
+	ErrObjectNotFound = errors.New("object not found")
+	// ErrSnapNameConflict is generated when a requested CSI snap name already exists on RBD but with
+	// different properties, and hence is in conflict with the passed in CSI volume name
+	ErrSnapNameConflict = errors.New("snapshot name conflict")
+	// ErrPoolNotFound is returned when pool is not found
+	ErrPoolNotFound = errors.New("pool not found")
+)
 
-// NewErrSnapNameConflict returns a ErrSnapNameConflict error when CSI snap name already exists.
-func NewErrSnapNameConflict(name string, err error) ErrSnapNameConflict {
-	return ErrSnapNameConflict{name, err}
+type errPair struct {
+	first  error
+	second error
 }
 
-// ErrPoolNotFound is returned when pool is not found
-type ErrPoolNotFound struct {
-	Pool string
-	Err  error
+func (e errPair) Error() string {
+	return fmt.Sprintf("%v: %v", e.first, e.second)
 }
 
-// Error returns the error string for ErrPoolNotFound.
-func (e ErrPoolNotFound) Error() string {
-	return e.Err.Error()
+// Is checks if target error is wrapped in the first error.
+func (e errPair) Is(target error) bool {
+	return errors.Is(e.first, target)
 }
 
-// Unwrap returns the encapsulated error.
-func (e ErrPoolNotFound) Unwrap() error {
-	return e.Err
+// Unwrap returns the second error.
+func (e errPair) Unwrap() error {
+	return e.second
 }
 
-// NewErrPoolNotFound returns a new ErrPoolNotFound error.
-func NewErrPoolNotFound(pool string, err error) ErrPoolNotFound {
-	return ErrPoolNotFound{pool, err}
+// JoinErr combines two errors. Of the returned error, Is() follows the first
+// branch, Unwrap() folllows the second branch.
+func JoinErr(e1, e2 error) error {
```

Comment:
this function can be renamed to JoinError

Reply:
But shouldn't it be `JoinErrors` then? And I would prefer the shorter version, because all errors also have to start with `Err`. Do you folks have a style guide regarding that, or is it your personal preference? What do others think?

Reply:
Yeah JoinErrors makes more sense , it was my personal preference,
Cc @humblec @nixpanic

Reply:
For most functions, `..Error..()` and `..Errors..()` seem to be used. `Err..` as constants/variables is common, but would not apply here.